.github/workflows/tests.yaml

```diff
@@ -0,0 +1,19 @@
+name: Run Tests
+on:
+  push:
+jobs:
+  run-tests:
+    # using `main` as the ref will keep your workflow up-to-date
+    uses: hashicorp/vault-workflows-common/.github/workflows/tests.yaml@main
+
+  run-tests-race:
+    name: "Run Tests Race"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version-file: .go-version
+          cache: true
+      - name: Run Tests (Race)
+        run: make testrace
```

Comment:
Doing both `make test` and `make testrace` since the original CircleCI workflow was testing this